### PR TITLE
changelog prep for v1.0.0

### PR DESCRIPTION
**Pull request by tac0turtle**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,7 +22,6 @@
 - [#646](https://github.com/cosmos/iavl/pull/646) Remove the `orphans` from the storage
 - [#777](https://github.com/cosmos/iavl/pull/777) Don't return errors from ImmutableTree.Hash, NewImmutableTree
 - [#815](https://github.com/cosmos/iavl/pull/815) `NewMutableTreeWithOpts` was removed in favour of accepting options via a variadic in `NewMutableTree`
-- [#815](https://github.com/cosmos/iavl/pull/815) `NewImmutableTreeWithOpts` is removed in favour of accepting options via a variadic in `NewImmutableTree`
```
Comment (julienrbrt) [inline CHANGELOG.md:25]: Why is this removed? This was done too.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## Unreleased
+## v1.0.0 (August 23, 2023)
```
Comment (julienrbrt) [inline CHANGELOG.md:3]: A final without https://github.com/cosmos/iavl/pull/805 ?

Comment (tac0turtle) [reply CHANGELOG.md:3]: true missed that pr, ill wait for it

File: CHANGELOG.md (hunk 1 above)
Comment (tac0turtle) [reply CHANGELOG.md:25]: the line above it is the same other than the word `is`

Comment (tac0turtle) [reply CHANGELOG.md]: ah missed that part. thanks

**Review by julienrbrt — APPROVED** [review]
lgtm, maybe we should mention it is an rc
